appinventor/blocklyeditor/src/field_lexical_variable.js

```diff
@@ -866,12 +924,25 @@ Blockly.LexicalVariable.makeLegalIdentifier = function(ident) {
 //    (1) add prefix argument,
 //    (2) handle local declaration statements/expressions, and
 //    (3) treat prefixes correctly when they're used.
+// TODO: I think this breaks the single responsibility principle. Consider
+//   splitting it into two functions, since this is only called internally.
+// TODO: Might be good to split the recursive part from the non-recursive part.
+//   I.E. Drop the env var and create a separate recursive function that
+//   takes it.
 Blockly.LexicalVariable.referenceResult = function (block, name, prefix, env) {
+  // TODO: Pretty sure block should never be null.
```

Comment:
Yes, it is possible due to the recursion. See for example line 1007 where we pass the result of `block.getInputTargetBlock('STACK')`. If there isn't a stack this will be null.

Reply:
Gosh golly, thank you for checking my logic! Removed.